projects/cm_mcu/CommandLineTask.c

```diff
@@ -156,62 +156,88 @@ typedef struct __attribute__((packed)) {
 
 extern struct dev_i2c_addr_t pm_addrs_dcdc[];
 
+static BaseType_t reset_all(int argc, char **argv, char *m)
+{
+  static int which = 0;
+  static int page_d = 0;
+  for (; which < N_PM_ADDRS_DCDC; ++which) {
+    for (; page_d < 2; ++page_d) { // for reading two pages per device
+      bool reset = false;
+      reset = true;
+      uint8_t sn[32];
+      snapdump(&pm_addrs_dcdc[which], page_d, sn, reset);
+    }
+  }
+  return pdFALSE;
+}
+
 static BaseType_t snapshot(int argc, char **argv, char *m)
 {
   _Static_assert(sizeof(snapshot_t) == 32, "sizeof snapshot_t");
   int copied = 0;
-  int page = strtol(argv[1], NULL, 10); // which LGA08D
-  int which = page / 10;
-  page = page % 10;
-  if (page < 0 || page > 1) {
-    copied += snprintf(m + copied, SCRATCH_SIZE - copied, "%s: page %d must be between 0-1\r\n",
-                       argv[0], page);
-    return pdFALSE;
+  if (strlen(argv[1]) < 3) {
+    char cpargv[2];
+    strcpy(cpargv, argv[1]);
```

Comment:
`strcpy` should never be used as it easily leads to overflow; use `strncpy` instead.

Reply:
no longer use strcpy or such.